### ISSUE TITLE
podman compose suggests using ---replace but doesn't have such an option

### ISSUE THREAD

**Issue by axelsimon** [containers/podman#24368]
### Issue Description

If you stop a stack started with `podman compose up` from a compose file, but don't destroy it, when you try to start it again podman will rightly complain that the container names are already taken.
And will say :
> Error: creating container storage: the container name "shiny-container" is already in use by 015180d2e9f53abd16bfe5386961a0c90fc99c6b354eca62a21480e8e9e3e17c. You have to remove that container to be able to reuse that name: that name is already in use, or use --replace to instruct Podman to do so.

But `--replace` is not a compose flag, currently it only exists for  podman create, pod create, run (according to [replace.md](https://github.com/containers/podman/blob/main/docs/source/markdown/options/replace.md)).

However. compose knows `--force`.

So either compose should respond to `--replace` or it should suggest using `--force` instead 🙂 

### Steps to reproduce the issue

Steps to reproduce the issue
1.start a stack with `podman compose up` from a compose file
2. ctrl+C
3. start the stack again with `podman compose up`
4. fails as container name is already in use, suggests using `--replace`
5. `podman compose --replace` : exit status 2

### Describe the results you received

"use --replace to instruct Podman" to remove the existing container

### Describe the results you expected

`--replace` is recognised by `podman compose`

Or, at least, the output should suggest: "use --force to instruct Podman" to remove the existing container

### podman info output

host:
  arch: amd64
  buildahVersion: 1.33.8
  cgroupControllers:
  - memory
  - pids
  cgroupManager: systemd
  cgroupVersion: v2
  conmon:
    package: conmon-2.1.10-1.el9.x86_64
    path: /usr/bin/conmon
    version: 'conmon version 2.1.10, commit: 3ea3d7f99779af0fcd69ec16c211a7dc3b4efb60'
  cpuUtilization:
    idlePercent: 98.75
    systemPercent: 0.55
    userPercent: 0.7
  cpus: 2
  databaseBackend: sqlite
  distribution:
    distribution: rocky
    version: "9.4"
  eventLogger: journald
  freeLocks: 2026
  hostname: rocky9.novalocal
  idMappings:
    gidmap:
    - container_id: 0
      host_id: 1000
      size: 1
    - container_id: 1
      host_id: 100000
      size: 65536
    uidmap:
    - container_id: 0
      host_id: 1000
      size: 1
    - container_id: 1
      host_id: 100000
      size: 65536
  kernel: 5.14.0-427.28.1.el9_4.x86_64
  linkmode: dynamic
  logDriver: journald
  memFree: 965513216
  memTotal: 3837399040
  networkBackend: netavark
  networkBackendInfo:
    backend: netavark
    dns:
      package: aardvark-dns-1.10.0-3.el9_4.x86_64
      path: /usr/libexec/podman/aardvark-dns
      version: aardvark-dns 1.10.0
    package: netavark-1.10.3-1.el9.x86_64
    path: /usr/libexec/podman/netavark
    version: netavark 1.10.3
  ociRuntime:
    name: crun
    package: crun-1.14.3-1.el9.x86_64
    path: /usr/bin/crun
    version: |-
      crun version 1.14.3
      commit: 1961d211ba98f532ea52d2e80f4c20359f241a98
      rundir: /run/user/1000/crun
      spec: 1.0.0
      +SYSTEMD +SELINUX +APPARMOR +CAP +SECCOMP +EBPF +CRIU +YAJL
  os: linux
  pasta:
    executable: ""
    package: ""
    version: ""
  remoteSocket:
    exists: false
    path: /run/user/1000/podman/podman.sock
  security:
    apparmorEnabled: false
    capabilities: CAP_CHOWN,CAP_DAC_OVERRIDE,CAP_FOWNER,CAP_FSETID,CAP_KILL,CAP_NET_BIND_SERVICE,CAP_SETFCAP,CAP_SETGID,CAP_SETPCAP,CAP_SETUID,CAP_SYS_CHROOT
    rootless: true
    seccompEnabled: true
    seccompProfilePath: /usr/share/containers/seccomp.json
    selinuxEnabled: true
  serviceIsRemote: false
  slirp4netns:
    executable: /usr/bin/slirp4netns
    package: slirp4netns-1.2.3-1.el9.x86_64
    version: |-
      slirp4netns version 1.2.3
      commit: c22fde291bb35b354e6ca44d13be181c76a0a432
      libslirp: 4.4.0
      SLIRP_CONFIG_VERSION_MAX: 3
      libseccomp: 2.5.2
  swapFree: 0
  swapTotal: 0
  uptime: 382h 18m 32.00s (Approximately 15.92 days)
  variant: ""
plugins:
  authorization: null
  log:
  - k8s-file
  - none
  - passthrough
  - journald
  network:
  - bridge
  - macvlan
  - ipvlan
  volume:
  - local
registries:
  search:
  - registry.access.redhat.com
  - registry.redhat.io
  - docker.io
store:
  configFile: /home/rocky/.config/containers/storage.conf
  containerStore:
    number: 9
    paused: 0
    running: 2
    stopped: 7
  graphDriverName: overlay
  graphOptions: {}
  graphRoot: /home/rocky/.local/share/containers/storage
  graphRootAllocated: 20245913600
  graphRootUsed: 8258228224
  graphStatus:
    Backing Filesystem: xfs
    Native Overlay Diff: "true"
    Supports d_type: "true"
    Supports shifting: "false"
    Supports volatile: "true"
    Using metacopy: "false"
  imageCopyTmpDir: /var/tmp
  imageStore:
    number: 7
  runRoot: /run/user/1000/containers
  transientStore: false
  volumePath: /home/rocky/.local/share/containers/storage/volumes
version:
  APIVersion: 4.9.4-rhel
  Built: 1728894259
  BuiltTime: Mon Oct 14 08:24:19 2024
  GitCommit: ""
  GoVersion: go1.21.13 (Red Hat 1.21.13-4.el9_4)
  Os: linux
  OsArch: linux/amd64
  Version: 4.9.4-rhel

### Podman in a container

No

### Privileged Or Rootless

None

### Upstream Latest Release

No

### Additional environment details

Additional environment details

### Additional information

Additional information like issue happens only occasionally or issue happens with a particular architecture or on a particular setting

**Comment by Luap99** [comment]
podman compose is just a wrapper around calling `docker-compose` or `podman-compose`. In you case you are using podman-compose and this is just calling podman run ... so it has no idea on how it was called and such this is impossible to fix.

You would need to create an issue on the [podman-compose](https://github.com/containers/podman-compose/) repo (which is not maintained by the podman maintainers) and ask them to implement such a flag

**Comment by vrothberg** [comment]
I think we could still massage the error message a bit.  Clients going through the REST API would need to be instructed how to replace a container.  Mentioning the --replace flag in those cases can cause confusion.  But I don't have a good proposal how to rephrase the error message to account for both use cases: Podman CLI and Podman REST API.  Maybe after another coffee :^)

**Comment by Luap99** [comment]
>  Clients going through the REST API would need to be instructed how to replace a container. Mentioning the --replace flag in those cases can cause confusion

But we don't! We only add this message in the cli case and podman-compose just runs cli commands.
https://github.com/containers/podman/blob/2f6fca6edc5c0d4e0a1d758a14c9de7014cd0561/cmd/podman/root.go#L667-L668

There is no replace option in the API, clients already know they have to delete and create a again.

**Comment by vrothberg** [comment]
In the case of `podman compose` the line is blurry I guess as it's a CLI command to essentially guide a client talk to the API.

Couldn't we make this error message conditional and only add the --replace note when the command is not compose?

**Comment by Luap99** [comment]
> Couldn't we make this error message conditional and only add the --replace note when the command is not compose?

Why? It is

**Comment by vrothberg** [comment]
Thanks for elaborating, @Luap99!

I was under the impression this would also happen with docker-compose.  But since (as you explained) podman-compose uses podman-run, there's nothing podman could do.

**Comment by dvishal485** [comment]
Yeah, `podman compose` is a wrapper around `podman run`, so issue doesn't relate to `podman`.
But still would like to answer that you may be probably looking for `--force-recreate` flag.

As per the docs, it recreate containers even if their configuration and image haven't changed.
